src/pages/menuPage/MenuPage.jsx

```diff
@@ -2,105 +2,98 @@ import './MenuPage.css';
+    const handleCategoryClick = (category) => {
+        setSelectedCategory(category);
+        setVisibleItemsCount(6);
     };
 
-    handleSeeMoreClick = () => {
-        this.setState(prevState => ({
-            visibleItemsCount: prevState.visibleItemsCount + 6
-        }));
+    const handleSeeMoreClick = () => {
+        setVisibleItemsCount(prevCount => prevCount + 6);
     };
 
-    render() {
-        const { selectedCategory, menuItems, categories, loading, error, visibleItemsCount } = this.state;
-        const { onAddToCart } = this.props;
-
-        const filteredItems = menuItems.filter(item => item.category === selectedCategory);
-        const itemsToShow = filteredItems.slice(0, visibleItemsCount);
-
-        const isSeeMoreVisible = filteredItems.length > visibleItemsCount;
+    const filteredItems = menuItems.filter(item => item.category === selectedCategory);
+    const itemsToShow = filteredItems.slice(0, visibleItemsCount);
+    const isSeeMoreVisible = filteredItems.length > visibleItemsCount;
 
-        return (
-            <div className="menu">
-                <h1>Browse our menu</h1>
-                <p>
-                    Use our menu to place an order online, or <Tooltip text="phone" tooltipText="+1-234-567-8901"/> our
-                    store to place a pickup order. Fast and fresh food.
-                </p>
-                <div className="categories">
-                    {categories.map((category, index) => (
-                        <Button
-                            key={index}
-                            label={category}
-                            onClick={() => this.handleCategoryClick(category)}
-                            variant={selectedCategory === category ? "primary" : "secondary"}
+    return (
+        <div className="menu">
+            <h1>Browse our menu</h1>
+            <p>
+                Use our menu to place an order online, or <Tooltip text="phone" tooltipText="+1-234-567-8901" /> our
+                store to place a pickup order. Fast and fresh food.
+            </p>
+            <div className="categories">
+                {categories.map((category, index) => (
+                    <Button
+                        key={index}
+                        label={category}
+                        onClick={() => handleCategoryClick(category)}
+                        variant={selectedCategory === category ? "primary" : "secondary"}
+                    />
+                ))}
+            </div>
+            <div className="menu-items">
+                {loading ? (
+                    <p>Loading...</p>
+                ) : error ? (
+                    <p>Error: {error}</p>
+                ) : (
+                    itemsToShow.map(item => (
+                        <MenuItem
+                            key={item.id}
+                            item={item}
+                            onAddToCart={onAddToCart}
                         />
```

Comment:
It's not good practices to use nested ternary operator. It worth thinking about reorganizing code